src/ucp/core/ucp_thread.h

```diff
@@ -82,6 +80,23 @@ typedef struct ucp_mt_lock {
         }                                                               \
     }
 
+/**
+ * Always lock on ucp_context.
+ */
+#define UCP_CTX_THREAD_IS_REQUIRED(_lock_ptr)   _THREAD_IS_REQUIRED(_lock_ptr)
+#define UCP_CTX_THREAD_LOCK_INIT(_lock_ptr)     _THREAD_LOCK_INIT(_lock_ptr)
+#define UCP_CTX_THREAD_LOCK_FINALIZE(_lock_ptr) _THREAD_LOCK_FINALIZE(_lock_ptr)
+#define UCP_CTX_THREAD_CS_ENTER(_lock_ptr)      _THREAD_CS_ENTER(_lock_ptr)
+#define UCP_CTX_THREAD_CS_EXIT(_lock_ptr)       _THREAD_CS_EXIT(_lock_ptr)
+
+#if ENABLE_MT
+
+#define UCP_THREAD_IS_REQUIRED(_lock_ptr)   _THREAD_IS_REQUIRED(_lock_ptr)
+#define UCP_THREAD_LOCK_INIT(_lock_ptr)     _THREAD_LOCK_INIT(_lock_ptr)
+#define UCP_THREAD_LOCK_FINALIZE(_lock_ptr) _THREAD_LOCK_FINALIZE(_lock_ptr)
+#define UCP_THREAD_CS_ENTER(_lock_ptr)      _THREAD_CS_ENTER(_lock_ptr)
+#define UCP_THREAD_CS_EXIT(_lock_ptr)       _THREAD_CS_EXIT(_lock_ptr)
```

Comment:
looks like these 2 are not needed anymore

Reply:
Actually other lines not needed too :). Thanks. Workers uses `async.h` macros.